autopilot/contractor/hostscore.go

```diff
@@ -60,6 +60,12 @@ func hostScore(cfg api.AutopilotConfig, h api.Host, expectedRedundancy float64)
 //     A 2x ratio will already cause the score to drop to 0.16 and a 3x ratio causes
 //     it to drop to 0.05.
 func priceAdjustmentScore(hostCostPerPeriod types.Currency, cfg api.ContractsConfig) float64 {
+	// return early if the allowance or amount of hosts is zero, avoiding a
+	// division by zero panic below.
+	if cfg.Allowance.IsZero() || cfg.Amount == 0 {
```

Comment:
I'd lean towards returning the smallest float here. Otherwise people might misinterpret a host as good when in fact they shouldn't trust the score they are receiving at all.

Could you also extend `TestHostScore` with 2 calls where one time the allowance is zero and one time the `cfg.Amount` is zero? That way we make sure none of the functions panic.